### PR TITLE
New Atom.json

### PR DESCRIPTION
**Pull request by jophab**
What does your Instant Answer do?
Atom text editor shortcuts
What is the data source for your Instant Answer? (Provide a link if possible)
http://www.shortcutworld.com/en/linux/Atom-(text-editor)_1.0.html
Why did you choose this data source?
It covers mor topics
Are there any other alternative (better) data sources?
maybe
Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)
developers
Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?
nope
Are you having any problems? Do you need our help with anything?
nope
What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
image

---
https://duck.co/ia/view/atom_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata post coming soon

**Comment by codenirvana** [comment]
This will create conflicts with [github-atom.json](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/cheat_sheets/json/github-atom.json)

**Comment by moollaza** [comment]
> This will create conflicts with github-atom.json

:+1: 

@jophab did you make this because the existing one uses OSX keys?

If so, we could probably swap out the OSX specific keys for windows keys for now. Long term we hope to solve the platform specific keys in a better way, so you an switch between them, and default to the OS you're using

**Comment by codenirvana** [comment]
> Long term we hope to solve the platform specific keys in a better way, so you an switch between them, and default to the OS you're using

@moollaza We need a ```PC | Mac``` switch for the cross platform keyboard templates.

+1 if we can detect the OS to solve this problem.

**Comment by gautamkrishnar** [comment]
@moollaza why the Travis CI checks failing??

**Comment by jophab** [comment]
@moollaza Actually I was unaware of that file.
I just checked whether there is an existing  atom.json file.
I found none.So I made this.
Cant we edit the aliases of existing atom cheat sheet to resolve conflicts?

File: share/goodie/cheat_sheets/json/atom.json [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			"key": "[Ctrl] [Up Arrow] / [Down Arrow]"
+		}, {
+			"val": "Toggle comment line",
+			"key": "[Ctrl] [/]"
+		}, {
+			"val": "New line below",
+			"key": "[Ctrl] [Enter]"
+		}, {
+			"val": "Indent / outdent selected lines",
+			"key": "[Ctrl] [[] / []]"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/atom.json]: @moollaza how to display square brackets?
Now its having some problem.

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/atom.json]: You can display it by using `\\[` or `\\]`

Comment (jophab) [inline share/goodie/cheat_sheets/json/atom.json]: @gautamkrishnar thanks daa :+1:

**Comment by gautamkrishnar** [comment]
@moollaza updated github-atom.json to mac cheatsheet... :+1: 
Will make more changes if needed :smile:

**Comment by moollaza** [comment]
@jophab please make an IA page for this PR. It will then autodeploy to Beta for testing + review

**Comment by moollaza** [comment]
You can go to https://duck.co/ia/new_ia and click the "skip this step" text, then provide the link to this PR

File: share/goodie/cheat_sheets/json/atom.json [hunk 2]
```diff
@@ -0,0 +1,143 @@
+{
+	"id": "atom_cheat_sheet",
+	"name": "Atom",
+	"description": "Keyboard shortcuts for Atom text Editor",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/atom.json]: no need to capitalized `editor`

File: share/goodie/cheat_sheets/json/atom.json [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	"metadata": {
+		"sourceName": "ShortcutWorld",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/atom.json]: looks like you've used hard tabs? Please switch to 4-space soft tabs :+1:

**Comment by jophab** [comment]
@moollaza  what is the perl module,template and dependencies asked in IA page?

**Comment by mintsoft** [comment]
@jophab For cheatsheets there aren't any external perl dependencies

**Comment by moollaza** [comment]
>what is the perl module,template and dependencies asked in IA page?

@jophab @mintsoft when filling out the IA Page for a cheat sheet, the "Perl Module" for all cheat sheets is `DDG::Goodie::CheatSheets`, "Template" can be left blank,  "Tab Name" should be `Cheat Sheet` -- we have documentation coming to explain this

/cc @duckduckgo/community-leaders @duckduckgo/duckduckhack-mentors -- please take note of this if anyone else asks!  As mentioned, there should be docs changes coming to explain this

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=atom for windows help)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/atom_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/atom.json [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+		"View": [{
+			"val": "Increase / decrease text size",
+			"key": "[Ctrl] [Shift] [=] / [-]"
+		}, {
+			"val": "Reset text size",
+			"key": "[Ctrl] [0]"
+		}, {
+			"val": "Toggle fullscreen",
+			"key": "[F11]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/atom.json]: No brackets needed here :+1:

**Comment by moollaza** [comment]
@jophab Thanks! This LGTM :+1: (please fix the brackets in a new PR)

**Comment by gautamkrishnar** [comment]
@moolaza i can't find any field named "Tab Name" in new ia page?

**Comment by moollaza** [comment]
@jophab congrats! It's live: https://duckduckgo.com/?q=atom%20for%20windows%20help&ia=answer

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by jophab** [comment]
@moollaza The link is not working, ie the cheatsheet is not triggerred.
screenshot from 2016-01-24 14 56 13

**Comment by GuiltyDolphin** [comment]
@jophab [works fine](https://duckduckgo.com/?t=lm&q=atom+cheat+sheet&ia=cheatsheet) for me.

**Comment by jophab** [comment]
@GuiltyDolphin Nope.
What you saw is  the cheatsheet for atom mac.
This cheat sheet is for windows and linux atom. :+1:

**Comment by GuiltyDolphin** [comment]
@jophab My bad, yeah `atom for windows cheat sheet` doesn't trigger for me.

**Comment by moollaza** [comment]
@jophab this PR removed your Bible Books Cheat Sheet from the repo.

You ***need*** to make sure you're working from a clean branch, which only contains the commits necessary for the files you're adding/modifying.

**Pull request merged**